services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -1613,3 +1619,89 @@ specSparUserMigration = do
         ssoToUidSpar tid ssoid
 
       liftIO $ mbUserId `shouldBe` Just memberUid
+
+specReAuthSsoUserWithPassword :: SpecWith TestEnv
+specReAuthSsoUserWithPassword =
+  describe "Re-auth for SSO users" $
+    it "password user that was upgraded to SAML" $ do
+      -- user has been invited via TM and has a password
+      env <- ask
+      (owner, tid) <- call (createUserWithTeam (env ^. teBrig) (env ^. teGalley))
+      email <- randomEmail
+      user <- call $ inviteAndRegisterUser (env ^. teBrig) owner tid email
+      -- user adds a client
+      cId <- addClientInternal (env ^. teBrig) (userId user) (defNewClient PermanentClientType [prekey] lPrekey)
+      checkNumClients (env ^. teBrig) (userId user) 1
+      -- attempt to delete the client without password fails
+      deleteClient (env ^. teBrig) (userId user) cId Nothing 403
+      -- attempt to delete the client with wrong password fails
+      deleteClient (env ^. teBrig) (userId user) cId (Just "wrong password") 403
+      -- idp is created
+      SampleIdP idpmeta _privkey _ _ <- makeSampleIdPMetadata
+      apiVersion <- view teWireIdPAPIVersion
+      idp <- call $ callIdpCreate apiVersion (env ^. teSpar) (Just owner) idpmeta
+      -- then user gets upgraded to scim and gets saml credentials
+      tok <- registerScimToken tid (Just (idp ^. idpId))
+      _ <- listUsers tok (Just (filterBy "externalId" (fromEmail email)))
+      -- attempt to delete the client with wrong password still fails
+      deleteClient (env ^. teBrig) (userId user) cId (Just "wrong password") 403
```

Comment:
that's unexpected (by me), but makes as much sense as completely ignoring the password.